src/main/java/common/bankarskiSistem/service/UserService.java

```diff
@@ -36,48 +45,51 @@ public User updateUser(User user) {
     }
 
     // CREATE user
-    public User saveUser(User user) {
+    public User saveUser(User user) throws EntityAlreadyExistsException {
         if(user == null)
             throw new NullPointerException("Null user");
         User existingUser
-                = userRepository.findById(user.getPersonalId())
+                = userRepository.findByPersonalId(user.getPersonalId())
                 .orElse(null);
-        if (existingUser == null)
+        if (existingUser == null) {
+            if(!user.getBankAccounts().isEmpty()){
+                for(BankAccount account : user.getBankAccounts()){
+                    account.setUser(user);
+                }
+            }
             return userRepository.save(user);
+        }
 
-        throw new NullPointerException("This user already exists!");
+        throw new EntityAlreadyExistsException("This user already exists!");
     }
 
     // DELETE user by personal id (jmbg)
-    public User deleteUserByPersonalId(String id) {
+    public User deleteUserByPersonalId(String id) throws EntityNotFoundException {
         if(id == null)
             throw new NullPointerException("Null personal id");
         User existingUser
-                = userRepository.findById(id)
+                = userRepository.findByPersonalId(id)
                 .orElse(null);
         if (existingUser == null)
-            throw new NullPointerException("No such user exists!");
+            throw new EntityNotFoundException("User not found!");
 
-        userRepository.deleteById(id);
-        return  existingUser;
+        userRepository.deleteByPersonalId(id);
+        return existingUser;
     }
 
     //CREATE BANK ACCOUNT for user
-    public BankAccount createBankAccount(User user, BankAccount bankAccount) {
-        if(user == null)
-            throw new NullPointerException("Null user");
+    public BankAccount createBankAccount(BankAccount bankAccount) throws EntityAlreadyExistsException {
         if(bankAccount == null)
             throw new NullPointerException("Null bank account");
-        User existingUser
-                = userRepository.findById(user.getPersonalId())
-                .orElse(null);
+        User existingUser = bankAccount.getUser();
+        log.info(bankAccount.getUser().toString());
         if (existingUser == null)
-            throw new NullPointerException("This bank account already exists!");
+            throw new EntityAlreadyExistsException("This bank account already exists!");
 
-        List<BankAccount> accounts = user.getBankAccounts();
+        List<BankAccount> accounts = existingUser.getBankAccounts();
         accounts.add(bankAccount);
```

Comment:
Ovde bi trebalo i bankAccountu dodati usera. Najbolja praksa za to je da se u modelu User napravi metod `addAccount` koji ce dodati account i accountu dodati usera `setUser(this)`

https://thorben-janssen.com/best-practices-for-many-to-many-associations-with-hibernate-and-jpa/